### PR TITLE
Skip aliasing correction for `lift_fresh`.

### LINKED ISSUE #111506: XLA Tensor creation fails on functionalization inside dynamo.
### 🐛 Describe the bug

Minimal reproducible program:

```python
import torch
import torch_xla.core.xla_model as xm

device = xm.xla_device()

def foo():
    return torch.tensor([0.0], device=device)

compiled = torch.compile(backend="openxla")(foo)
tensor = compiled()
```

```python
  File "torch/_subclasses/functional_tensor.py", line 304, in __torch_dispatch__
    return return_and_correct_aliasing(func, args, kwargs, outs_wrapped)
  File "torch/utils/_python_dispatch.py", line 364, in return_and_correct_aliasing
    _correct_storage_aliasing(func, schema_info, args, (out,) if not isinstance(out, tuple) else out)
  File "torch/utils/_python_dispatch.py", line 251, in _correct_storage_aliasing
    alias_non_inplace_storage(args[arg_idx], outs[return_idx])
  File "torch/utils/_python_dispatch.py", line 234, in alias_non_inplace_storage
    torch.ops.aten.set_.source_Storage_storage_offset(ret, arg.untyped_storage(), ret.storage_offset(), ret.shape)
  File "torch/_ops.py", line 499, in __call__
    return self._op(*args, **kwargs or {})
torch._dynamo.exc.BackendCompilerFailed: backend='openxla' raised:
RuntimeError: Attempted to set the storage of a tensor on device "xla:0" to a storage on different device "lazy:0".  This is no longer allowed; the devices must match.

While executing %tensor : [num_users=1] = call_function[target=torch.tensor](args = ([0.0],), kwargs = {device: xla:0})
Original traceback:
  File "examples/bug-device.py", line 7, in foo
    return torch.tensor([0.0], device=device)
```

### Versions

PyTorch version: 2bf3ca1be759460cf9fbf011d96d3246001361e9 (Oct 4)
PyTorch/XLA version: c9a132484fb89bfdc9c602ada7bd8a3cec0db1aa (Oct 3)

cc @bdhirsh @ezyang

### PR DESCRIPTION
**Pull request by ysiraichi**
Fix: #111506

This PR skips aliasing correction on `lift_fresh` calls. Reasoning is: although unlifted and lifted tensors are technically aliases, they are from different levels of abstraction (`FunctionalTensorWrapper` and `XLATensor`).

cc @bdhirsh @ezyang

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Can you add a test for this please?

File: torch/_subclasses/functional_tensor.py [hunk 1]
```diff
@@ -308,10 +308,17 @@ def unwrap(x):
...
+            # If no outputs are our functional subclass, then don't try to fix up aliasing
+            not any(
+                isinstance(x, FunctionalTensor)
+                for x in pytree.tree_flatten(outs_wrapped)[0]
+            )
+            # Since lift_fresh lifts its argument into a functional tensor, we can skip the
+            # aliasing correction step. Otherwise, we would be setting the storage of a
+            # lifted tensor to that of an unlifted tensor.
+            # Ref: https://github.com/pytorch/pytorch/issues/111506
+            or func == torch.ops.aten.lift_fresh.default
```
Comment (bdhirsh) [inline torch/_subclasses/functional_tensor.py:321]: Hmm, I think that if the func here is `lift_fresh`, you might need to manually wrap the output in a `FunctionalTensor()`.

File: test/test_functionalization.py [hunk 2]
```diff
@@ -1639,6 +1639,14 @@ def forward(self, arg0_1):
     return add""")
         self.assertEqual(fx_g_cpp.code.strip(), fx_g.code.strip())
 
+    def test_python_functionalization_lift_fresh_storage(self):
+        unlifted = torch.tensor([0.0])
```
Comment (bdhirsh) [inline test/test_functionalization.py:1643]: Hmm, can you make this test a bit more complicated? e.g.:
```
def f(x):
    tmp = torch.tensor([0.0])
    return x + tmp
```

That will at least convince me of whether the change for `lift_fresh` below works properly :). There's an invariant that if an output to an op is an `at::functionalization::FunctionalTensorWrapper`, then python functionalization needs to wrap the output in a python `FunctionalTensor` subclass.

File: torch/_subclasses/functional_tensor.py (hunk 1 above)
Comment (ysiraichi) [reply torch/_subclasses/functional_tensor.py:321]: But, we are already returning the wrapped version of the outputs, right!?

https://github.com/pytorch/pytorch/blob/21330e5ba19b4f22afbd26168eb72bcd8586071d/torch/_subclasses/functional_tensor.py#L298

**Comment by ysiraichi** [comment]
@bdhirsh I think this PR is ready for another round of reviews. Could you take a look at it when you have time?

File: torch/_subclasses/functional_tensor.py (hunk 1 above)
Comment (bdhirsh) [reply torch/_subclasses/functional_tensor.py:321]: oh yep you're right!

**Review by bdhirsh — APPROVED** [review]
lgtm!

**Pull request closed without merging**